### DISCUSSION TITLE
framework adapter vs native usage of webcomponents

### DISCUSSION THREAD

**Discussion by deleonio** [telekom/scale#800] [General]
Hi @acstll,

why you said that?

https://github.com/telekom/scale/issues/615#issuecomment-1009715918

I have observed that using the web components directly is not as convenient as using the framework adapters. DX is very important for us. For this we rely on auto-completion and intellisense.

The seamless integration into react or solidjs development is impressive.

When using the web components in angular or vue, do you also have this DX comfort? How does he read the component schemata in the IDE?

**Comment by christopherGdynia** [comment]
@deleonio 

It took us some time to figure it out with @acstll for Solid and React, I will take a look a Angular, what we can do about it.

**Reply by W1ckh3ad** [reply]
@deleonio @acstll 

We won't get any native intellisence for html in angular, don't know about vue yet.
But if we write an ```vscode extension``` for ```scale``` ```html intellisence```.
We would cover raw ```html``` and ```angular``` with it.

What do you think about it?

**Reply by deleonio** [reply]
Thanks for the feedback.  Auto-completion is not really possible with own template languages ​​like Angular and Vue. Only with complex extensions.

My point is your response regarding my PR on template issue. (https://github.com/telekom/scale/issues/615#issuecomment-1009715918)

I don't know if DX works well using native web components?  Are you using Scale in Angular and Vue directly or through the Stenci Output Adapter?

I tried to configure the components.d.ts with a completely new Angular 13.2 project so that you have auto-completion in the HTML. Unfortunately I still failed.

**Comment by acstll** [comment]
Hey @deleonio, @W1ckh3ad!

I'm really sorry for my lack of reply (I don't get any notifications for this Discussions feature, gonna check my settings).

This topic is really important for us, I need to find some time to post a proper and detailed answer. I'll get back to you this week.

Thanks for asking @deleonio

**Reply by acstll** [reply]
I obviously did not find the time and then forgot about this (really sorry). The topic is still open… @deleonio were you maybe able to find a solution?

**Comment by christopherGdynia** [comment]
@deleonio I totally forget about this aswell, 

I have not used scale in a big project with angular, I just created in this #707 a starter, I worked there with out wrapper. 
I have nerver worked with vue.
In react I am using the wrapper.
